### PR TITLE
chore: update dev and system test images

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Update dependencies (dev container and system test image):

* update to Debian 12 (bookworm)
* install mosquitto >= 2.0.18 as it fixes some bugs with the bridge that thin-edge.io is reliant on
* bump `robotframework-devicelibrary` version (for system tests)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by reubenmiller** [comment]
@albinsuresh I tested out the dev container definition in this image and it built without any errors.

File: docs/src/contribute/BUILDING.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ You can use any OS to build from source (below has been tested on Ubuntu, but we
 
 Our recommended setup and required tools are:
 
-* Ubuntu 20.04 or Debian 10.9 (Buster)
+* Ubuntu 20.04 or Debian 12 (Bookworm)
```
Comment (albinsuresh) [inline docs/src/contribute/BUILDING.md:15]: We may have to increase that minimum Ubuntu version as well, as its inbuilt repositories have an older version of mosquitto: `1.9.1` IIRC, unless we expect even the users to configure the sid repo and update to the exact version that we recommend.

**Review by albinsuresh — APPROVED** [review]

File: docs/src/contribute/BUILDING.md (hunk 1 above)
Comment (reubenmiller) [reply docs/src/contribute/BUILDING.md:15]: though I guess that is only regarding building the project and not specifically using it

Comment (reubenmiller) [reply docs/src/contribute/BUILDING.md:15]: But yes I think we need to update all of the build instructions to also use the newer way of doing things using zigbuild and the just(file) recipes.

Comment (albinsuresh) [reply docs/src/contribute/BUILDING.md:15]: Aah okay, just noticed that it's `BUILDING.md` and not `supported-platforms.md`, where we document the officially supported versions, which still refers to Ubuntu 20.04.

**Pull request merged**